app/scripts/directives/payment.js

```diff
@@ -64,21 +64,88 @@ angular.module('confRegistrationWebApp')
+                    }, {
+                      validate: cruPayments.card.validate.knownType,
+                      errorMessage: 'This card type is not recognized.'
+                    }, {
+                      validate: cruPayments.card.validate.typeLength,
+                      get errorMessage() {
+                        var cardNumber = currentPayment.creditCard.number;
+                        var cardType = cruPayments.card.info.type(cardNumber);
+                        var expectedLength = cruPayments.card.info.expectedLengthForType(cardNumber) || [];
+                        return 'This is an invalid ' + cardType + ' number. ' +
+                               'It should have ' + expectedLength.join(' or ') + ' digits.';
+                      }
+                    }, {
+                      validate: cruPayments.card.validate.checksum,
+                      errorMessage: 'This card number is invalid. At least one digit was entered incorrectly.'
+                    }
+                  ],
+                  cvvNumber: [
+                    {
+                      validate: required,
+                      errorMessage: 'You must enter the security code.'
+                    }, {
+                      validate: cruPayments.cvv.validate.minLength,
+                      errorMessage: 'The security code must be at least 3 digits.'
+                    }, {
+                      validate: cruPayments.cvv.validate.maxLength,
+                      errorMessage: 'The security code cannot be more than 4 digits.'
+                    }
+                  ],
+                  expirationMonth: [
+                    {
+                      validate: required,
+                      errorMessage: 'You must choose an expiration month.'
+                    }, {
+                      validate: function (expirationMonth) {
+                        return cruPayments.expiryDate.validate.month(expirationMonth, currentPayment.creditCard.expirationYear);
+                      },
+                      errorMessage: 'Your credit card is expired or you selected the wrong month or year.'
+                    }
+                  ],
+                  expirationYear: [
+                    {
+                      validate: required,
+                      errorMessage: 'You must choose an expiration year.'
+                    }, {
+                      validate: cruPayments.expiryDate.validate.year,
+                      errorMessage: 'Your credit card is expired or you selected the wrong year.'
+                    }
+                  ]
+                };
+
+                // Run all of the validations
+                _.forEach(validations, function (validators, field) {
+                  // Validate this field
+                  validators.forEach(function (validator) {
+                    if (!validator.validate(currentPayment.creditCard[field])) {
+                      paymentErrors.push(gettext(validator.errorMessage));
```

Comment:
I researched it, and, yes. you are correct. `gettext` doesn't actually do any translation, whereas `gettextCatalog.getString` does. Both forms mark the string as needing translation. For this reason, I switched to using the latter.